### PR TITLE
readme/contributing: sponsorship notes and minor tweaks

### PR DESCRIPTION
**Pull request by davidhewitt**
I added a little bit about sponsorship to the README and contributing notes, including a link to my recently-set-up Github sponsors page for those who would like to tip me for the time spent working on PyO3. I hope that the community finds it ok if I add a little bit of self-promotion here.

At the same time I updated a few bits and pieces that I think makes the content easier to navigate.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,22 +2,17 @@
...
 
 * User Guide: [stable](https://pyo3.rs) | [main](https://pyo3.rs/main)
 
 * API Documentation: [stable](https://docs.rs/pyo3/) |  [main](https://pyo3.rs/main/doc)
 
-* Contributing Notes: [github](https://github.com/PyO3/pyo3/blob/main/Contributing.md)
-
-* Architecture Guide: [github](https://github.com/PyO3/pyo3/blob/master/Architecture.md)
-
-A comparison with rust-cpython can be found [in the guide](https://pyo3.rs/main/rust_cpython.html).
```
Comment (davidhewitt) [inline README.md:19]: I wanted to make this section at the top of the README shorter, so I moved the contributing notes link into a badge, put the architecture guide link down in the new contributing section below, and removed the rust-cpython comparison note from here.

(Users who are interested in understanding the differences can still find the comparison in the guide.)

**Review by kngwyu — APPROVED** [review]
LGTM, and congrats for getting started to be sponsored 🎉

**Comment by messense** [comment]
I'd like to propose to also add a [FUNDING.yml](https://docs.github.com/en/github/administering-a-repository/displaying-a-sponsor-button-in-your-repository) to display the sponsor button which is much more discoverable.

**Comment by davidhewitt** [comment]
I'd obviously like that, however I think it's not my place to make such a decision. While I am the most active contributor at the moment, I think the contributors who were here before me (e.g. yourself, @kngwyu and @konstin) have the rights to decide whether we should add that. 

A sponsor button on the PyO3 repo would look like the one here: https://github.com/jonschlinkert/kind-of

The words "sponsor this project" above my profile and then the "Sponsor" button. It's not 100% accurate as any sponsorship money pledged through that button would be going directly to me rather than to all of the maintainers. The Github UI does make that clear at least: clicking on the "Sponsor" button would take you through to my sponsorship page.

**Comment by konstin** [comment]
As long as this only show up on pyo3/pyo3, I'll leave this for @kngwyu to decide

**Comment by kngwyu** [comment]
As I said in the Github org discussion, I'm personally OK. The only thing I take care of is the impression to future contributors. If we have an organization funding account, I guess it can attract future contributors. However, it can be too tiresome for @davidhewitt to manage both org and personal funding, so a personal account is also OK, I think.

**Pull request merged**